File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    /*initialize input*/
    double init_val;
    do
    {
        init_val = get_double("Change owed: ");
    }
    while (init_val < 0);
    
    /*quarters*/
    int val = init_val * 100;
    int tot = val / 25;
    val %= 25;
    
    /*dimes*/
    tot += val / 10;
    val %= 10;
    
    /*nickels*/
    tot += val / 5;
    
    /*pennies*/
    tot += val % 5;
    
    printf("%i\n", tot);
}

File: pset6/cash/cash.py
while True:
    try:
        print("Change owed: ", end = "")
        c = float(input())
        if c >= 0:
            break
    except ValueError:
        continue

c = int(c * 100)
tot = c // 25
c %= 25
tot += c // 10
c %= 10
tot += (c // 5 + c % 5)
print(tot)